src/shared/components/Orgs/OrgList.tsx

```diff
@@ -1,83 +1,152 @@
 import * as React from 'react';
-import { Input, Empty } from 'antd';
-import OrgItem, { OrgItemProps } from './OrgItem';
-import AnimatedList from '../Animations/AnimatedList';
-
+import { Spin } from 'antd';
+import { OrgResponseCommon, OrganizationList } from '@bbp/nexus-sdk';
+import { PaginatedList } from '../../utils/types';
+import Search from 'antd/lib/input/Search';
+import OrgItem from './OrgItem';
+import { AccessControl, useNexus } from '@bbp/react-nexus';
 import './Orgs.less';
-import { Organization } from '@bbp/nexus-sdk-legacy';
+import { DEFAULT_UI_SETTINGS } from '../../utils/consts';
+import InfiniteScroll from '../Animations/InfiniteScroll';
+import { FetchableState } from '../../store/reducers/utils';
 
-export interface OrgListProps {
-  orgs: OrgItemProps[];
-  busy?: boolean;
-  error?: { message: string; name: string };
-  onOrgClick?(org: Organization): void;
-  onOrgEdit?(label: string): void;
-  paginationSettings?: { total: number; from: number; pageSize: number };
-  onPaginationChange?: (page: number, pageSize?: number) => void;
+export interface OrgListContainerProps {
+  pageSize?: number;
+  onOrgClick?(orgLabel: string): void;
+  onOrgEdit?(orgLabel: string): void;
+  createOrg?(): void;
 }
 
-const Search = Input.Search;
+export const OrgsListContainer: React.FunctionComponent<
+  OrgListContainerProps
+> = props => {
+  const {
+    pageSize = DEFAULT_UI_SETTINGS.pageSizes.orgsListPageSize,
+    onOrgClick,
+    onOrgEdit,
+    createOrg,
+  } = props;
+  const [{ from, size, label, deprecated }, setQuery] = React.useState({
+    size: pageSize,
+    from: 0,
+    label: '',
+    deprecated: false,
+  });
+  const [fetchablePaginatedList, setPaginatedList] = React.useState<
+    FetchableState<PaginatedList<OrgResponseCommon>>
+  >({
+    error: null,
+    isFetching: true,
+    data: {
+      index: from,
+      total: 0,
+      results: [],
+    },
+  });
+
+  const { loading, error, data } = useNexus<OrganizationList>(
+    nexus =>
+      nexus.Organization.list({
+        from,
+        size,
+        label,
+        deprecated,
+      }),
+    [from, size, label, deprecated]
+  );
+
+  React.useEffect(() => {
```

Comment:
I'm not sure I get that bit. You already have all the data you need above. All you're doing in that `useEffect` is derive a new state by combining 2, without even changing any values. :thinking:

Reply:
Yeah this is one I struggled with. The problem is we only want to update the InfiniteScroll list when the response data comes, and not when the state changes, otherwise InfiniteScroll won't know if it's a new list and won't refresh itself. 

Maybe extending InfiniteScroll with a refresh function prop to force a refresh would be a way to avoid this.